cli/flox-rust-sdk/src/models/manifest/typed.rs

```diff
@@ -383,6 +383,35 @@ pub enum ManifestPackageDescriptor {
     StorePath(PackageDescriptorStorePath),
 }
 
+impl ManifestPackageDescriptor {
+    /// Check if two package descriptors should have the same resolution.
+    /// This is used to determine if a package needs to be re-resolved
+    /// in the presence of an existing lock.
+    ///
+    /// * Descriptors are resolved per system,
+    ///   changing the supported systems does not invalidate _existing_ resolutions.
+    /// * Priority is not used in resolution, so it is ignored.
+    pub(crate) fn is_from_custom_catalog(&self) -> bool {
+        use ManifestPackageDescriptor::*;
+        match self {
```

Comment:
Nit: when you only care about one variant of the enum, it's idiomatic to use an `if let` like this:
```rust
if let ManifestPackageDescriptor::Catalog(pkg) = self {
    // logic
} else {
    false
}
```